model/map/src/main/java/org/keycloak/models/map/user/MapCredentialValidationOutput.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.user;
+
+import org.keycloak.models.CredentialValidationOutput;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Output of credential validation
+ *
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class MapCredentialValidationOutput {
+
+    private final MapUserEntity authenticatedUser; // authenticated user.
```

Comment:
Just realized that this needn't be `MapUserEntity` in case of partial stores which do not necessarily return the entity but only an intermediate object which is turned into a `MapUserEntity` by mappers.

Could you please replace this with
```suggestion
public class MapCredentialValidationOutput<V> {

    private final V authenticatedUser; // authenticated user.
```

This can be safely cast to `MapUserEntity` in the [`getUserByCredentials` method](https://github.com/hmlnarik/keycloak/pull/23/files#diff-90c66f213572f75c2455b1b176eb7b16c3740209567dc7e5002253f882852f67R788) - partial store would by that time have it converted to `MapUserEntity` by the mappers.

Reply:
generified, see c0bf1d58f29eef149f9a4b6f69c364d5640c2c3e